temporal/api/cloud/cloudservice/v1/request_response.proto

```diff
@@ -357,8 +357,14 @@ message GetUserGroupsRequest {
     string page_token = 2;
     // Filter groups by the namespace they have access to - optional.
     string namespace = 3;
-    // Filter groups by their name - optional.
-    string group_name = 4;
+    // Filter groups by the display name - optional.
+    string display_name = 4;
+    // Filter groups by the group type - optional.
+    // Possible values: google-group
+    string group_type = 5;
+    // Filter groups by the google group email - optional.
+    // Only applicable when group_type filter is set to google group-type.
+    string google_group_email_address = 6;
```

Comment:
Can you do the same thing here as you did in the spec? Specifically can you have a `GoogleGroupFilter` that can contain filterable fields and even just its empty presence is like `group_type`? My concern is that you're going to add a new top-level field for every field inside every spec type, and they only work if they are set as the same time as something else which is confusing.

Reply:
Just updated and removed the type field.

> and even just its empty presence is like group_type
Proto3 does not distinguish between empty and unset scalars. 
https://github.com/protocolbuffers/protobuf/issues/1606#issuecomment-281832148
I don't think we can implement a type filter with just a googleGroupFilter.

Reply:
> Proto3 does not distinguish between empty and unset scalars.

That is unrelated to my concern now about top-level fields. Are you going to add a new top-level field for every field you want to filter on each group type? My suggestion is just making a a child object for all google group filter fields (and another for the next one) instead of moving them all top level. Same reason you have `UserGroupSpec.google_group.email_address` instead of `UserGroupSpec.google_group_email_address, you should have `GetUserGroupsRequest.google_group.email_address` instead of `GetUserGroupsRequest.google_group_email_address`.

Reply:
Got it. Let me do that.

Reply:
Got it, makes sense. Added a child object as suggested.